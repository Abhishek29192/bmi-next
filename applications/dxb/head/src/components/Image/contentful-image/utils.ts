import type { ImageLoaderProps } from "next/image";
import type { Data, Options, Props } from "./types";

export const typeToObjectFitMap: {
  [key in NonNullable<Data["type"]>]: Options["size"];
} = {
  Decorative: "cover",
  Descriptive: "contain"
};

export const getPosition = ({
  size,
  position,
  focalPoint,
  isMobile
}: Options & Pick<Data, "focalPoint">): Options["position"] => {
  if (position) {
    return position;
  }

  if (!isMobile && size === "cover" && focalPoint) {
    return `${focalPoint.x}% ${focalPoint.y}%`;
  }

  return "center";
};

export const loader = ({ src, width }: ImageLoaderProps) => {
  return `${src}?w=${width}&q=50`;
};

export const getSizes = (widths: Props["widths"]) =>
  `(max-width: 599px) ${widths[0]}px, (max-width: 719px) ${widths[1]}px, (max-width: 839px) ${widths[2]}px, (max-width: 1439px) ${widths[3]}px, ${widths[4]}px`;
