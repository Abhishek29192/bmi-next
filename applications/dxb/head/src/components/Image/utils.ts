import { Props } from "./types";
import type { Props as ContentfulImageProps } from "./contentful-image/types";

export const isContentfulImage = (
  props: Props
): props is ContentfulImageProps =>
  "__typename" in props && props.__typename === "ContentfulImage";
