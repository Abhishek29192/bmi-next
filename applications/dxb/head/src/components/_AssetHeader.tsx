import React, { useState } from "react";
import {
  AnchorLink,
  Button,
  Dialog,
  Icon,
  Tooltip
} from "@bmi-digital/components";
import ClickAwayListener from "@material-ui/core/ClickAwayListener";
import InfoIcon from "@material-ui/icons/Info";
import { microCopy } from "../constants/microCopies";
import RichText from "./RichText";
import { useSiteContext } from "./Site";
import styles from "./styles/DocumentTechnicalTableResults.module.scss";
import { PIMDocumentData, PIMLinkDocumentData } from "./types/PIMDocumentBase";

const AssetHeader = ({
  assetType
}: {
  assetType: PIMDocumentData["assetType"] | PIMLinkDocumentData["assetType"];
}) => {
  const { name, code, description } = assetType;
  const [isTooltipActive, setIsTooltipActive] = useState<boolean>(false);
  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
  const { getMicroCopy } = useSiteContext();

  return (
    <>
      {description && (
        <Dialog open={isDialogOpen} onCloseClick={() => setIsDialogOpen(false)}>
          <Dialog.Title>
            {name} ({code})
          </Dialog.Title>
          <Dialog.Content>
            <RichText document={description} />
          </Dialog.Content>
          <Dialog.Actions
            confirmLabel={getMicroCopy(microCopy.DIALOG_CLOSE)}
            onConfirmClick={() => setIsDialogOpen(false)}
          />
        </Dialog>
      )}
      <ClickAwayListener
        onClickAway={() => {
          setIsTooltipActive(false);
        }}
      >
        <Tooltip
          PopperProps={{
            disablePortal: true
          }}
          title={
            <>
              {name}
              {description && (
                <>
                  <br />
                  <AnchorLink
                    onClick={() => setIsDialogOpen(true)}
                    color="white"
                  >
                    {getMicroCopy(microCopy.GLOBAL_LEARN_MORE)}...
                  </AnchorLink>
                </>
              )}
            </>
          }
          open={isTooltipActive}
          interactive={!!description}
        >
          <div>
            <Button
              isIconButton
              variant="text"
              size={42}
              onClick={() =>
                setIsTooltipActive(
                  (prevIsTooltipActive) => !prevIsTooltipActive
                )
              }
            >
              <Icon source={InfoIcon} className={styles["tooltip-icon"]} />
            </Button>
          </div>
        </Tooltip>
      </ClickAwayListener>
    </>
  );
};

export default AssetHeader;
