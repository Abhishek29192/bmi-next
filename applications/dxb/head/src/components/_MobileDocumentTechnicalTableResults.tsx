import React from "react";
import {
  Accordion,
  AccordionSummaryProps,
  Button,
  Clickable,
  ClickableProps,
  Icon,
  IconButtonProps,
  iconMap
} from "@bmi/components";
import withGTM from "../utils/google-tag-manager";
import AssetHeader from "./_AssetHeader";
import { Data as AssetTypeData } from "./AssetType";
import styles from "./styles/DocumentTechnicalTableResults.module.scss";
import { Format } from "./types";
import { PIMDocumentData, PIMLinkDocumentData } from "./types/PIMDocumentBase";

interface Props {
  documentsByProduct: [string, (PIMDocumentData | PIMLinkDocumentData)[]][];
  assetTypes: AssetTypeData[];
  fileIconsMap: Record<Format, React.ComponentType>;
}

const MobileDocumentTechnicalTableResults = ({
  documentsByProduct,
  assetTypes,
  fileIconsMap
}: Props) => {
  const GTMAccordionSummary = withGTM<AccordionSummaryProps>(Accordion.Summary);
  const GTMClickable = withGTM<ClickableProps>(Clickable);
  const GTMButton = withGTM<IconButtonProps>(Button);

  return (
    <div className={styles["accordion-div"]}>
      <Accordion>
        {documentsByProduct.map(([productName, assets], index) => {
          const key = `${
            assets.find(({ product }) => product.code)?.product.code
          }-${index}`;
          return (
            <Accordion.Item key={key}>
              <GTMAccordionSummary
                gtm={{
                  id: "selector-accordion1",
                  label: productName,
                  action: "Selector â€“ Accordion"
                }}
              >
                {assets.length > 0 ? assets[0].product.name : productName}
              </GTMAccordionSummary>
              {assetTypes.map((assetType, index) => {
                const filteredAssets = assets.filter(
                  ({ assetType: { id } }) => id === assetType.id
                );

                return filteredAssets.map((asset, assetIndex) => (
                  <Accordion.Details
                    key={`${productName}-asset-${asset.id}-${index}-${assetIndex}`}
                    className={styles["accordion-details"]}
                  >
                    <div className={styles["icon-container"]}>
                      {asset.__typename !== "PIMLinkDocument" ? (
                        <Icon
                          source={
                            fileIconsMap[asset.format] || iconMap.External
                          }
                          className={styles["format-icon"]}
                        />
                      ) : (
                        <Icon
                          source={iconMap.External}
                          className={styles["external-link-icon"]}
                        />
                      )}
                    </div>
                    <div className={styles["icon-container"]}>
                      {assetType.code}
                    </div>
                    <div className={styles["info-icon-container"]}>
                      <AssetHeader assetType={assetType} />
                    </div>
                    <div className={styles["download-icon-container"]}>
                      {asset.__typename !== "PIMLinkDocument" ? (
                        <GTMClickable
                          model="download"
                          href={asset.url}
                          download={asset.title}
                          gtm={{
                            id: "download1",
                            label: "Download",
                            action: asset.url
                          }}
                        >
                          <Icon
                            source={iconMap.Download}
                            className={styles["all-files-icon"]}
                          />
                        </GTMClickable>
                      ) : (
                        <GTMButton
                          isIconButton
                          accessibilityLabel="Download"
                          variant="text"
                          action={{
                            model: "htmlLink",
                            href: asset.url,
                            target: "_blank",
                            rel: "noopener noreferrer"
                          }}
                          gtm={{
                            id: "download1",
                            label: "Download",
                            action: asset.url
                          }}
                        >
                          <Icon
                            source={iconMap.Download}
                            className={styles["all-files-icon"]}
                          />
                        </GTMButton>
                      )}
                    </div>
                  </Accordion.Details>
                ));
              })}
            </Accordion.Item>
          );
        })}
      </Accordion>
    </div>
  );
};

export default MobileDocumentTechnicalTableResults;
