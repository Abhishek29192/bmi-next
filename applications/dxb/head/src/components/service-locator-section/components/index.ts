// istanbul ignore file: doesn't hold any logic
export * from "./CompanyDetails";
export * from "./IntegratedLinkCard";
export * from "./ServiceLocationBlock/SearchLocationBlock";
export * from "./ServiceLocatorChips/ServiceLocatorChips";
export * from "./ServiceLocatorMap";
export * from "./ServiceLocatorResultList/ServiceLocatorResultList";
export * from "./useScrollTo";
