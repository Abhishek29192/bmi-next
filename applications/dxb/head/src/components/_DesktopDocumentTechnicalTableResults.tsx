import React, { useContext } from "react";
import classnames from "classnames";
import {
  Button,
  Clickable,
  ClickableProps,
  DownloadList,
  DownloadListContext,
  Icon,
  IconButtonProps,
  iconMap,
  Table
} from "@bmi/components";
import axios from "axios";
import { useGoogleReCaptcha } from "react-google-recaptcha-v3";
import { downloadAs } from "../utils/client-download";
import withGTM from "../utils/google-tag-manager";
import { useConfig } from "../contexts/ConfigProvider";
import { microCopy } from "../constants/microCopies";
import AssetHeader from "./_AssetHeader";
import { useSiteContext } from "./Site";
import { Data as AssetTypeData } from "./AssetType";
import styles from "./styles/DocumentTechnicalTableResults.module.scss";
import { Format } from "./types";
import { PIMDocumentData, PIMLinkDocumentData } from "./types/PIMDocumentBase";
import createAssetFileCountMap, {
  AssetUniqueFileCountMap,
  generateFilenameByRealFileName,
  generateFileNamebyTitle
} from "./DocumentFileUtils";

interface Props {
  documentsByProduct: [string, (PIMDocumentData | PIMLinkDocumentData)[]][];
  assetTypes: AssetTypeData[];
  fileIconsMap: Record<Format, React.ComponentType>;
}

const DesktopDocumentTechnicalTableResults = ({
  documentsByProduct,
  assetTypes,
  fileIconsMap
}: Props) => {
  const {
    config: { documentDownloadEndpoint }
  } = useConfig();
  const { getMicroCopy } = useSiteContext();
  const { list } = useContext(DownloadListContext);
  const { executeRecaptcha } = useGoogleReCaptcha();

  const GTMClickable = withGTM<ClickableProps>(Clickable);
  const GTMButton = withGTM<IconButtonProps>(Button);

  const singleDocument = (asset: PIMDocumentData | PIMLinkDocumentData) =>
    asset.__typename !== "PIMLinkDocument" ? (
      <GTMClickable
        model="download"
        href={asset.url}
        download={asset.title}
        gtm={{
          id: "download1",
          label: "Download",
          action: asset.url
        }}
      >
        <Icon
          source={fileIconsMap[asset.format]}
          className={styles["format-icon"]}
        />
      </GTMClickable>
    ) : (
      <GTMButton
        isIconButton
        accessibilityLabel="Download"
        variant="text"
        action={{
          model: "htmlLink",
          href: asset.url,
          target: "_blank",
          rel: "noopener noreferrer"
        }}
        gtm={{
          id: "download1",
          label: "Download",
          action: asset.url
        }}
        className={styles["external-download-button"]}
        disableTouchRipple={true}
      >
        <Icon
          source={iconMap.External}
          className={styles["external-link-icon"]}
        />
      </GTMButton>
    );

  const multipleDocuments = (
    assets: (PIMDocumentData | PIMLinkDocumentData)[]
  ) => {
    const downloadMultipleFiles = async () => {
      try {
        if (!documentDownloadEndpoint) {
          throw Error(
            "`GATSBY_DOCUMENT_DOWNLOAD_ENDPOINT` missing in environment config"
          );
        }
        const [currentTime] = new Date()
          .toJSON()
          .replace(/-|:|T/g, "")
          .split(".");
        let zipFileName = `BMI_${currentTime}.zip`;
        if (assets.length > 0 && assets[0].product && assets[0].assetType) {
          zipFileName = `${assets[0].product?.name} ${assets[0].assetType.name}.zip`;
        }

        const token = await executeRecaptcha();
        const pimDocumentAssets: PIMDocumentData[] = assets.filter(
          (asset): asset is PIMDocumentData =>
            asset.__typename === "PIMDocument"
        );
        const assetFileCountMap: AssetUniqueFileCountMap =
          createAssetFileCountMap(pimDocumentAssets);
        const documents = pimDocumentAssets.map((asset, index) => ({
          href: asset.url,
          name:
            asset.realFileName && asset.realFileName !== ""
              ? generateFilenameByRealFileName(assetFileCountMap, asset, index)
              : generateFileNamebyTitle(
                  assetFileCountMap,
                  asset.title,
                  asset.extension,
                  index
                )
        }));
        const response = await axios.post(
          documentDownloadEndpoint,
          { documents: documents },
          { responseType: "text", headers: { "X-Recaptcha-Token": token } }
        );
        await downloadAs(response.data.url, zipFileName);
      } catch (error) {
        console.error("Download multiple documents", error); // eslint-disable-line
      }
    };

    return (
      <GTMButton
        isIconButton
        accessibilityLabel="Download"
        variant="text"
        action={{
          model: "default"
        }}
        onClick={downloadMultipleFiles}
        gtm={{
          id: "download1",
          label: "Download",
          action: JSON.stringify(
            Object.values(assets).map((asset) => asset.url)
          )
        }}
        disableTouchRipple={true}
        className={styles["external-download-button"]}
      >
        <Icon source={iconMap.FileZIP} className={styles["format-icon"]} />
      </GTMButton>
    );
  };

  return (
    <div className={styles["table-div"]}>
      <Table rowBgColorPattern="none">
        <Table.Head>
          <Table.Row className={styles["header-row"]}>
            <Table.Cell>
              {getMicroCopy(microCopy.DOCUMENT_LIBRARY_HEADERS_PRODUCT)}
            </Table.Cell>
            {assetTypes.map((assetType, index) => {
              return (
                <Table.Cell
                  key={`asset-type-${assetType.id}-${index}`}
                  className={styles["asset-type-cell"]}
                >
                  {assetType.code}
                  <AssetHeader assetType={assetType} />
                </Table.Cell>
              );
            })}
            <Table.Cell className={styles["all-files-header"]}>
              <span className={styles["all-files-header-wrapper"]}>
                <Icon
                  source={iconMap.Download}
                  className={styles["all-files-icon"]}
                />
                <span>
                  {getMicroCopy(microCopy.DOCUMENT_LIBRARY_HEADERS_ALL_FILES)}
                </span>
              </span>
            </Table.Cell>
          </Table.Row>
        </Table.Head>
        <Table.Body>
          {documentsByProduct.map(([productName, assets], index) => {
            const key = `${
              assets.find(({ product }) => product.code)?.product.code
            }-${index}`;
            const hasOnlyExternalAssets = assets.every(
              (asset) => asset.__typename === "PIMLinkDocument"
            );
            return (
              <Table.Row
                key={key}
                className={classnames(styles["row"], {
                  // eslint-disable-next-line security/detect-object-injection
                  [styles["row--checked"]]: !!list[key]
                })}
              >
                <Table.Cell>
                  {assets.length > 0 ? assets[0].product.name : productName}
                </Table.Cell>
                {assetTypes.map((assetType, index) => {
                  const filteredAssets = assets.filter(
                    ({ assetType: { id } }) => id === assetType.id
                  );

                  if (!filteredAssets.length) {
                    return (
                      <Table.Cell
                        key={`${productName}-missing-asset-${index}`}
                        className={styles["align-center"]}
                      >
                        <Icon
                          source={iconMap.Cross}
                          className={styles["no-document-icon"]}
                        />
                      </Table.Cell>
                    );
                  }

                  return (
                    <Table.Cell
                      key={`${productName}-asset-${assetType.id}`}
                      className={styles["align-center"]}
                    >
                      {filteredAssets.length === 1
                        ? singleDocument(filteredAssets[0])
                        : multipleDocuments(filteredAssets)}
                    </Table.Cell>
                  );
                })}
                <Table.Cell className={styles["align-center"]}>
                  {!hasOnlyExternalAssets && (
                    <DownloadList.Checkbox
                      name={key}
                      maxLimitReachedLabel={getMicroCopy(
                        microCopy.DOCUMENTS_DOWNLOAD_MAX_REACHED
                      )}
                      ariaLabel={`${getMicroCopy(
                        microCopy.DOCUMENT_LIBRARY_DOWNLOAD
                      )} ${productName}`}
                      value={assets.filter(
                        ({ __typename }) => __typename !== "PIMLinkDocument"
                      )}
                      fileSize={assets.reduce((acc, curr) => {
                        if (curr.__typename === "PIMLinkDocument") {
                          return 0;
                        }
                        return acc + (curr.fileSize || 0);
                      }, 0)}
                    />
                  )}
                </Table.Cell>
              </Table.Row>
            );
          })}
        </Table.Body>
      </Table>
    </div>
  );
};

export default DesktopDocumentTechnicalTableResults;
