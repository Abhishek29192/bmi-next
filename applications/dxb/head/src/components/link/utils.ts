import { Link as GatsbyLink } from "gatsby";
import { getPathWithCountryCode } from "../../utils/path";
import { DataTypeEnum } from "./types";
import type { Data } from "./types";
import type { AnchorLinkActionProps } from "@bmi-digital/components/anchor-link";
import type { ButtonActionProps } from "@bmi-digital/components/button";
import type { Data as PageInfoData } from "../PageInfo";
import type { Data as PromoData } from "../Promo";

export const getCTA = (
  data: Pick<PromoData, "cta"> | Pick<PageInfoData, "path">,
  countryCode: string,
  linkLabel?: string
): AnchorLinkActionProps | null => {
  if ("cta" in data) {
    if (!data.cta) {
      return null;
    }

    return toAnchorLinkActionProps(data.cta, countryCode);
  }

  const { path } = data;
  if (!path) {
    return null;
  }

  const to = getPathWithCountryCode(countryCode, path).replace(/\/+/gi, "/");
  return {
    to: to,
    component: GatsbyLink,
    gtm: { id: "cta-click1", action: to, label: linkLabel }
  };
};

const internalHosts = process.env.GATSBY_INTERNAL_HOSTS?.split(",") || [
  "www.bmigroup.com"
];

export const isExternalUrl = (url: string): boolean => {
  try {
    const linkUrl = new URL(url);
    return !internalHosts.includes(linkUrl.host);
  } catch (e) {
    return false;
  }
};

const checkUrlAction = (url: string): boolean => {
  const actionUrls = ["mailto:", "tel:", "callto:"];
  return actionUrls.some((actionUrl) => url.startsWith(actionUrl));
};
export const toAnchorLinkActionProps = (
  link: Data,
  countryCode?: string
): AnchorLinkActionProps => {
  if (link.asset?.file?.url) {
    return {
      href: link.asset.file.url,
      download: true,
      external: true,
      gtm: {
        id: "cta-click1",
        action: link.asset.file.url,
        label: link.label
      }
    };
  }

  if (link.linkedPage?.path) {
    const to = getPathWithCountryCode(
      countryCode,
      link.linkedPage.path
    ).replace(/\/+/gi, "/");
    return {
      component: GatsbyLink,
      to,
      gtm: { id: "cta-click1", action: to, label: link.label }
    };
  }

  // TODO: DXB-7055 link.url _should_ be populated, but as any link can be used, it may not be
  return {
    href: link.url ?? undefined,
    external:
      (!!link.url && !checkUrlAction(link.url) && isExternalUrl(link.url)) ||
      undefined,
    gtm: { id: "cta-click1", action: link.url ?? undefined, label: link.label }
  };
};

export const toButtonActionProps = (
  link: Data,
  countryCode?: string,
  onClick?: ButtonActionProps["onClick"]
): ButtonActionProps => {
  if (link.type === DataTypeEnum.Visualiser) {
    return {
      onClick,
      gtm: {
        id: "cta-visualiser1",
        action: "visualiser",
        label: link.label
      }
    };
  }

  if (link.type === DataTypeEnum.Calculator) {
    return {
      onClick,
      gtm: {
        id: "cta-calculator1",
        action: "calculator",
        label: link.label
      }
    };
  }

  if (link.type === DataTypeEnum.Dialog) {
    return {
      onClick,
      gtm: { id: "cta-click1", action: link.type, label: link.label }
    };
  }

  return toAnchorLinkActionProps(link, countryCode);
};
