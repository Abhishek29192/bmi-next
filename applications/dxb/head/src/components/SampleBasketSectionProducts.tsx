import React from "react";
import { OverviewCard } from "@bmi/components";
import { Typography } from "@bmi/components";
import { Icon } from "@bmi/components";
import { Button } from "@bmi/components";
import { Remove } from "@material-ui/icons";
import { navigate } from "gatsby";
import { useMediaQuery, useTheme } from "@material-ui/core";
import { getPathWithCountryCode } from "../utils/path";
import { microCopy } from "../constants/microCopies";
import {
  ACTION_TYPES,
  Sample,
  useBasketContext
} from "../contexts/SampleBasketContext";
import styles from "./styles/SampleBasketSectionProducts.module.scss";
import { renderImage } from "./Image";
import { useSiteContext } from "./Site";
import { Classification } from "./types/pim";

interface Options {
  code: string;
  separator?: string;
  featureUnitRequired?: boolean;
}

export const getFeatures = (
  classifications: Classification[],
  options: Options
) => {
  const classification = classifications.find(
    (classification) =>
      classification.code.toLowerCase() === options.code.toLowerCase()
  );

  if (!classification) {
    return null;
  }

  const features = [
    ...new Set(
      classification?.features.flatMap((feature) =>
        feature.featureValues.map(({ value }) => value)
      )
    )
  ].filter((value) => value);

  const units =
    (options.featureUnitRequired &&
      classification?.features[0]?.featureUnit?.symbol) ||
    "";

  return `${features.join(options.separator || " | ")} ${units}`.trim();
};

const SampleBasketSectionProducts = () => {
  const { basketState, basketDispatch } = useBasketContext();
  const { getMicroCopy, countryCode } = useSiteContext();

  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("md"));

  const sampleCards = basketState.products.map((sample) => {
    const media = renderImage(
      {
        __typename: "ContentfulImage",
        type: "Descriptive",
        altText: sample.name,
        caption: { caption: sample.name },
        image: {
          file: {
            fileName: sample.name,
            url: sample.image
          }
        },
        focalPoint: { x: 0, y: 0 }
      },
      {
        className: styles["product-image"]
      }
    );

    const removeFromBasket = (payload: Sample) => {
      basketDispatch({
        type: ACTION_TYPES.BASKET_REMOVE,
        payload
      });
    };

    const handleRemove = (event: Event) => {
      event.stopPropagation();
      removeFromBasket(sample);
    };

    return (
      <OverviewCard
        className={styles["product-wrapper"]}
        key={sample.path}
        media={media}
        hasChildrenWithoutMargin
        onClick={() =>
          navigate(getPathWithCountryCode(countryCode, sample.path))
        }
        footer={
          !isMobile && (
            <Button
              className={styles["product-button"]}
              endIcon={<Remove />}
              onClick={handleRemove}
              variant="outlined"
            >
              {getMicroCopy(microCopy.PDP_OVERVIEW_REMOVE_SAMPLE)}
            </Button>
          )
        }
      >
        <div className={styles["product"]}>
          <div className={styles["product-description"]}>
            <Typography className={styles["product-title"]}>
              {sample.name}
            </Typography>
            <Typography className={styles["product-color"]}>
              {getFeatures(sample.classifications, {
                code: "appearanceAttributes"
              })}
            </Typography>
            <Typography className={styles["product-size"]}>
              {getFeatures(sample.classifications, {
                code: "measurements",
                separator: "x",
                featureUnitRequired: true
              })}
            </Typography>
          </div>
          {isMobile && (
            <Button
              className={styles["product-button"]}
              onClick={handleRemove}
              variant="outlined"
              isIconButton
            >
              <Icon source={Remove} />
            </Button>
          )}
        </div>
      </OverviewCard>
    );
  });

  return (
    <div className={styles["SampleBasketSectionProducts"]}>{sampleCards}</div>
  );
};

export default SampleBasketSectionProducts;
