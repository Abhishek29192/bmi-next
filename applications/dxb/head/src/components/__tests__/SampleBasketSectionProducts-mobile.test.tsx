import { ThemeProvider } from "@bmi/components";
import { render } from "@testing-library/react";
import React from "react";
import {
  BasketContextProvider,
  Sample
} from "../../contexts/SampleBasketContext";
import { local } from "../../utils/storage";
import createImage from "../../__tests__/helpers/ImageHelper";
import SampleBasketSectionProducts from "../SampleBasketSectionProducts";

const samples: Sample[] = [
  {
    name: "sample-1",
    code: "sample-1",
    path: "sample-1-details",
    colour: null,
    textureFamily: null,
    measurements: null,
    image: createImage().mainSource
  },
  {
    name: "sample-2",
    code: "sample-2",
    path: "sample-2-details",
    colour: "Red",
    textureFamily: "Matte",
    measurements: "10x20x30 mm",
    image: createImage().mainSource
  },
  {
    name: "sample-3",
    code: "sample-3",
    path: "sample-3-details",
    colour: "Black",
    textureFamily: null,
    measurements: null,
    image: createImage().mainSource
  }
];
jest.mock("@material-ui/core/useMediaQuery", () => () => true);
describe("SampleBasketSectionProducts component render correctly on mobile devices", () => {
  beforeAll(() => {
    jest.spyOn(local, "getItem").mockReturnValue(JSON.stringify(samples));
  });

  it("renders correctly for mobile", () => {
    const { container } = render(
      <ThemeProvider>
        <BasketContextProvider>
          <SampleBasketSectionProducts />
        </BasketContextProvider>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
