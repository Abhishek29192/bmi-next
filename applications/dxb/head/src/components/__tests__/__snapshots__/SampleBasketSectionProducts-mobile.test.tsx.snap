// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleBasketSectionProducts component render correctly on mobile devices renders correctly for mobile 1`] = `
<div>
  <div
    class="css-1yyiycy"
  >
    <button
      class="MuiButtonBase-root OverviewCard-clickable SampleBasketSectionProducts-product-wrapper MuiBox-root css-rctto-MuiButtonBase-root"
      data-testid="overview-card-box"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-1"
          class="SampleBasketSectionProducts-product-image OverviewCard-headerPicture css-7mgsqw-Media"
          loading="lazy"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body OverviewCard-noFooter"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="SampleBasketSectionProducts-product"
            >
              <div
                class="SampleBasketSectionProducts-product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-1
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root Button-root iconButton Button-medium MuiIconButton-sizeLarge SampleBasketSectionProducts-product-button css-142qa0u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RemoveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13H5v-2h14v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-iicyj8"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root OverviewCard-clickable SampleBasketSectionProducts-product-wrapper MuiBox-root css-rctto-MuiButtonBase-root"
      data-testid="overview-card-box"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-2"
          class="SampleBasketSectionProducts-product-image OverviewCard-headerPicture css-7mgsqw-Media"
          loading="lazy"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body OverviewCard-noFooter"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="SampleBasketSectionProducts-product"
            >
              <div
                class="SampleBasketSectionProducts-product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-2
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  Red | Matte
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  10x20x30 mm
                </p>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root Button-root iconButton Button-medium MuiIconButton-sizeLarge SampleBasketSectionProducts-product-button css-142qa0u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RemoveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13H5v-2h14v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-iicyj8"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root OverviewCard-clickable SampleBasketSectionProducts-product-wrapper MuiBox-root css-rctto-MuiButtonBase-root"
      data-testid="overview-card-box"
      tabindex="0"
      type="button"
    >
      <div
        class="OverviewCard-main"
        data-testid="overview-card-main"
      >
        <img
          alt="sample-3"
          class="SampleBasketSectionProducts-product-image OverviewCard-headerPicture css-7mgsqw-Media"
          loading="lazy"
          src="http://localhost:8000/image-main-source.jpg"
          style="object-fit: contain; object-position: center;"
        />
        <div
          class="OverviewCard-body OverviewCard-noFooter"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 OverviewCard-title OverviewCard-noBrandLogo css-1x9euu8-MuiTypography-root"
            data-testid="overview-card-body-title"
          />
          <div
            class="OverviewCard-text OverviewCard-withoutMargin css-kl66ak"
            data-testid="overview-card-body-description"
          >
            <div
              class="SampleBasketSectionProducts-product"
            >
              <div
                class="SampleBasketSectionProducts-product-description"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-title css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  sample-3
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-color css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                >
                  Black
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 SampleBasketSectionProducts-product-size css-1o02bdp-MuiTypography-root"
                  data-testid="typography"
                />
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root Button-root iconButton Button-medium MuiIconButton-sizeLarge SampleBasketSectionProducts-product-button css-142qa0u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RemoveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13H5v-2h14v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-iicyj8"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
